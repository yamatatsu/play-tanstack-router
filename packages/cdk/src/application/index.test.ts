import * as cdk from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import * as lambda from "aws-cdk-lib/aws-lambda";
import * as synthetics from "aws-cdk-lib/aws-synthetics";
import { expect, test, vi } from "vitest";
import { ApplicationStack } from ".";

vi.spyOn(lambda.Code, "fromAsset").mockReturnValue(
	lambda.Code.fromInline("dummy") as unknown as lambda.AssetCode,
);
vi.spyOn(synthetics.Code, "fromAsset").mockReturnValue(
	synthetics.Code.fromInline("dummy") as unknown as synthetics.AssetCode,
);

test("Snapshot test", () => {
	const app = new cdk.App();

	const stack = new ApplicationStack(app, "Target", {
		envName: "test",
		prefix: "test-prefix",
	});

	const template = Template.fromStack(stack);
	expect(template.toJSON()).toMatchSnapshot();
});
