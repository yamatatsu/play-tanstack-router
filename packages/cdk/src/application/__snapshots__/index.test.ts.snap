// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot test 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "CognitotestprefixUserPool30525942": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "AccountRecoverySetting": {
          "RecoveryMechanisms": [
            {
              "Name": "verified_email",
              "Priority": 1,
            },
          ],
        },
        "AdminCreateUserConfig": {
          "AllowAdminCreateUserOnly": true,
        },
        "AutoVerifiedAttributes": [
          "email",
        ],
        "DeletionProtection": "INACTIVE",
        "DeviceConfiguration": {
          "ChallengeRequiredOnNewDevice": true,
          "DeviceOnlyRememberedOnUserPrompt": true,
        },
        "EmailConfiguration": {
          "EmailSendingAccount": "COGNITO_DEFAULT",
        },
        "EmailVerificationMessage": "The verification code to your new account is {####}",
        "EmailVerificationSubject": "Verify your new account",
        "Policies": {
          "PasswordPolicy": {
            "MinimumLength": 8,
            "RequireLowercase": false,
            "RequireNumbers": false,
            "RequireSymbols": false,
            "RequireUppercase": false,
            "TemporaryPasswordValidityDays": 7,
          },
        },
        "SmsVerificationMessage": "The verification code to your new account is {####}",
        "UsernameAttributes": [
          "email",
        ],
        "VerificationMessageTemplate": {
          "DefaultEmailOption": "CONFIRM_WITH_CODE",
          "EmailMessage": "The verification code to your new account is {####}",
          "EmailSubject": "Verify your new account",
          "SmsMessage": "The verification code to your new account is {####}",
        },
      },
      "Type": "AWS::Cognito::UserPool",
      "UpdateReplacePolicy": "Delete",
    },
    "CognitotestprefixUserPooltestprefixUserPoolClientA455AF53": {
      "Properties": {
        "AccessTokenValidity": 60,
        "AllowedOAuthFlowsUserPoolClient": false,
        "ClientName": "test-prefixApplication",
        "IdTokenValidity": 60,
        "PreventUserExistenceErrors": "ENABLED",
        "RefreshTokenValidity": 1440,
        "SupportedIdentityProviders": [
          "COGNITO",
        ],
        "TokenValidityUnits": {
          "AccessToken": "minutes",
          "IdToken": "minutes",
          "RefreshToken": "minutes",
        },
        "UserPoolId": {
          "Ref": "CognitotestprefixUserPool30525942",
        },
      },
      "Type": "AWS::Cognito::UserPoolClient",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
